app/models/notification_setting_state.rb

```diff
@@ -0,0 +1,29 @@
+# == Schema Information
+#
+# Table name: notification_setting_states
+#
+#  id                      :integer          not null, primary key
+#  is_toggled              :boolean          default(TRUE)
+#  created_at              :datetime         not null
+#  updated_at              :datetime         not null
+#  notification_setting_id :integer          not null, indexed
+#  user_id                 :integer          not null, indexed
+#
+# Indexes
+#
+#  index_notification_setting_states_on_notification_setting_id
+#                                        (notification_setting_id)
+#  index_notification_setting_states_on_user_id                  (user_id)
+#
+# Foreign Keys
+#
+#  fk_rails_057c83c713  (user_id => users.id)
+#  fk_rails_66e9ee5396  (notification_setting_id => notification_settings.id)
+#
+
+class NotificationSettingState < ActiveRecord::Base
```

Comment:
Why not use an ActiveRecord `enum` column for this instead of a separate `notification_setting_id` + table?  Also perhaps it'd be good to have multiple `toggled` columns for the different methods of notification?